### PR TITLE
nsis: Upgrade to v3.01

### PR DESCRIPTION
**Pull request by idleberg**

### CODE REVIEW

**Comment by mojca** [comment]
Since you submitting this update, I would like to ask you for a tiny favour. I noticed that the port depends on mingw , but I'm unable to figure out in what way exactly.

I just prepared ports for `i686-w64-mingw32-gcc` and `x86_64-w64-mingw32-gcc` and would like to know if nsis would work with those ports instead of the ancient mingw (which I would eventually like to get rid of).

I still need to commit those two ports though (they are currently [here](https://github.com/mojca/macports-user-mojca/tree/master/ports/cross)).

Adding @kurthindenburg.

**Comment by idleberg** [comment]
Maybe @kichik or @sredna can answer this

File: devel/nsis/Portfile [hunk 1]
```diff
@@ -23,11 +23,11 @@ distfiles           nsis-${version}-src.tar.bz2 \
                     nsis-${version}.zip
 
 checksums           nsis-3.0-src.tar.bz2 \
```
Comment (mojca) [inline devel/nsis/Portfile]: This won't work. You submitted version for 3.01, but then check the checksums for version 3.0.

`3.0` should be replaced by `${version}` for both files.

Did you actually test the port?

File: devel/nsis/Portfile [hunk 2]
```diff
@@ -22,10 +22,10 @@ master_sites        sourceforge
 distfiles           nsis-${version}-src.tar.bz2 \
                     nsis-${version}.zip
 
-checksums           nsis-3.0-src.tar.bz2 \
+checksums           nsis-3.01-src.tar.bz2 \
```
Comment (mojca) [inline devel/nsis/Portfile]: Please replace with (literally) `${version}` rather than the hardcoded version.

**Comment by mojca** [comment]
Thank you very much for the contribution and request for the upgrade.

Your patch was perfectly OK, but I took a closer look at the port and there was a lot that was wrong with it: wrong compiler being used, "outdated variables", ... so I committed 430c387d10ab1635c0a4768ed1a8ade4d5d6167e and 32ea7ed8e38722d82dda310bbc2b786058efbd19 which includes your patch.

I also checked what [HB](https://github.com/Homebrew/homebrew-core/blob/c7fb40864ccf9f14358f8639949f6cf3e39260f2/Formula/makensis.rb) did.

I blindly copied `STRIP=no` based on [this](https://sourceforge.net/p/nsis/bugs/1085/#c71b/3859) and https://github.com/Homebrew/legacy-homebrew/issues/28718. I guess that the other workaround with stdlib might be obsolete by now and the extra code might not be needed.

**Comment by mojca** [comment]
The build fails on 10.6 due to missing dependencies:

* https://trac.macports.org/ticket/53058

but I'll take a look later.

**Comment by idleberg** [comment]
If my memory serves me well, 3.0 didn't build on 10.6 either. It might have been an oversight, but can you verify that?

**Comment by mojca** [comment]
It does build. I now have access to the 10.6 machine. See the ticket listed above.

**Pull request closed without merging**